eng/pipelines/common/xplat-setup.yml

```diff
@@ -50,11 +50,7 @@ jobs:
         value: $(buildConfigUpper)
 
       - name: _runSmokeTestsOnlyArg
-        ${{ if ne(variables['nonPRBuild'], true) }}:
-          value: /p:RunSmokeTestsOnly=true
-        ${{ if eq(variables['nonPRBuild'], true) }}:
-          value: ''
-
+        value: '/p:RunSmokeTestsOnly=$(!isFullMatrix)'
```

Comment:
I don't think you can do: `!` here.

Reply:
would `$[ not($isFullMatrix) ]` work?

Reply:
That should work, but the syntax would be: `$[ not(variables.isFullMatrix) ]`